src/services/teams.js

```diff
@@ -0,0 +1,17 @@
+import { checkError, client } from './client';
+
+export async function fetchTeams() {
+  const response = await client.from('teams').select().order('name');
+  return checkError(response);
+}
+
+export async function fetchTeamById(id) {
+  const response = await client.from('teams').select('*, players(*)').match({ id }).single();
+  return checkError(response);
+}
+
+// did not write this alone!
+export async function fetchPlayersByTeam(id) {
```

Comment:
The fetchTeamById function already joins the players into the response, no need for a fetchPlayersByTeam here